### ISSUE TITLE
SC2154 is triggered on an all caps variable

### ISSUE THREAD

**Issue by danielcft** [koalaman/shellcheck#3074]
- Rule Id: SC2154
- My shellcheck version: 0.10.0
- It works on shellcheck.net - could not reproduce the issue there.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

echo "${SRC}"
```

#### Here's what shellcheck currently says:
```
In bug.sh line 3:
echo "${SRC}"
      ^----^ SC2154 (warning): SRC is referenced but not assigned.

For more information:
  https://www.shellcheck.net/wiki/SC2154 -- SRC is referenced but not assigned.
```

#### Here's what I wanted or expected to see:

I expect no warnings. 

The variable uses the all caps convention, so it should not be flagged.
The documentation on wiki states it explicitly.

from https://www.shellcheck.net/wiki/SC2154
> Note: This message only triggers for variables with lowercase characters in their name (foo and kFOO but not FOO) due to the standard convention of using lowercase variable names for unexported, local variables.

#### Additional findings

Running on latest version using the docker I can reproduce the issue
```
$ docker run --rm -v "$PWD:/mnt" koalaman/shellcheck:latest --version
ShellCheck - shell script analysis tool
version: v0.10.0-57-g0970187
license: GNU General Public License, version 3
website: https://www.shellcheck.net

$ docker run --rm -v "$PWD:/mnt" koalaman/shellcheck:latest bug.sh

In bug.sh line 3:
echo "${SRC}"
      ^----^ SC2154 (warning): SRC is referenced but not assigned.

For more information:
  https://www.shellcheck.net/wiki/SC2154 -- SRC is referenced but not assigned.
```

**Comment by danielcft** [comment]
Okay, I found this optional check in my `.shellcheckrc`

`check-unassigned-uppercase`

**Comment by danielcft** [comment]
For reference, I was not the first one being confused with this one https://github.com/koalaman/shellcheck/issues/2904
